api/security.go

```diff
@@ -381,3 +383,60 @@ func DeletePolicy(c *gin.Context) {
 	createResponse(c, *response)
 
 }
+
+func GetImageDeployments(c *gin.Context) {
+	imageDigest := c.Param("imageDigest")
+	releaseMap := make(map[string]bool)
+	kubeConfig, ok := GetK8sConfig(c)
+	if !ok {
+		return
+	}
+	client, err := helm.GetK8sConnection(kubeConfig)
+	if err != nil {
+		log.Errorf("Error getting K8s config: %s", err.Error())
+		c.JSON(http.StatusBadRequest, pkgCommmon.ErrorResponse{
+			Code:    http.StatusBadRequest,
+			Message: "Error getting K8s config",
+			Error:   err.Error(),
+		})
+		return
+	}
+	// Get all pods from cluster
+	pods, err := listPods(client, "", "")
+	if err != nil {
+		log.Errorf("Error getting pods from cluster: %s", err.Error())
+		c.JSON(http.StatusBadRequest, pkgCommmon.ErrorResponse{
+			Code:    http.StatusBadRequest,
+			Message: "Error getting pods from cluster",
+			Error:   err.Error(),
+		})
+		return
+	}
+	// Example status
+	//   - containerID: docker://08054c2cbf7ac842c1dc93929fa1b7a07e28641433fa21790c44408e2f897584
+	//    image: banzaicloud/pipeline:debug
+	//    imageID: docker://sha256:0e61330d23a5fd32dbf56c47808abc0a960248b4415746dfd5540b455384a6a0
+	for _, p := range pods {
+		for _, status := range p.Status.ContainerStatuses {
+			if getImageDigest(status.ImageID) == imageDigest {
+				releaseMap[p.Labels["release"]] = true
+			}
+		}
+		for _, status := range p.Status.InitContainerStatuses {
+			if getImageDigest(status.ImageID) == imageDigest {
+				releaseMap[p.Labels["release"]] = true
+			}
+		}
+	}
+	var releaseList []string
+	for k := range releaseMap {
+		releaseList = append(releaseList, k)
+	}
+
+}
+
+func getImageDigest(imageID string) string {
+	image := strings.Split(imageID, "@")
```

Comment:
Probably we should make some validation here on imageId, with regexp or something.

Reply:
added basic validation